ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/tokenize/SparseTokenizerModel.java

```diff
@@ -0,0 +1,136 @@
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.opensearch.ml.common.CommonValue.ML_MAP_RESPONSE_KEY;
+import static org.opensearch.ml.engine.utils.FileUtils.deleteFileQuietly;
+
+@Log4j2
+@Function(FunctionName.SPARSE_TOKENIZE)
+public class SparseTokenizerModel extends DLModel {
+    private HuggingFaceTokenizer tokenizer;
+
+    private Map<String, Float> idf;
+
+    @Override
+    public ModelTensorOutput predict(String modelId, MLInput mlInput) throws TranslateException {
+        MLInputDataset inputDataSet = mlInput.getInputDataset();
+        List<ModelTensors> tensorOutputs = new ArrayList<>();
+        TextDocsInputDataSet textDocsInput = (TextDocsInputDataSet) inputDataSet;
+        ModelResultFilter resultFilter = textDocsInput.getResultFilter();
+        for (String doc : textDocsInput.getDocs()) {
+            Output output = new Output(200, "OK");
+            Encoding encodings = tokenizer.encode(doc);
+            long[] indices = encodings.getIds();
+            List<ModelTensor> outputs = new ArrayList<>();
+            String[] tokens = Arrays.stream(indices)
+                    .distinct()
+                    .mapToObj(value -> new long[]{value})
+                    .map(value -> this.tokenizer.decode(value, true))
+                    .filter(s -> !s.isEmpty())
+                    .toArray(String[]::new);
+            Map<String, Float> tokenWeights = Arrays.stream(tokens)
+                    .collect(Collectors.toMap(
+                            token -> token,
+                            token -> idf.getOrDefault(token, 1.0f)
+                    ));
+            Map<String, ?> wrappedMap = Map.of(ML_MAP_RESPONSE_KEY, Collections.singletonList(tokenWeights));
+            ModelTensor tensor = ModelTensor.builder()
+                    .dataAsMap(wrappedMap)
+                    .build();
+            outputs.add(tensor);
+            ModelTensors modelTensorOutput = new ModelTensors(outputs);
+            output.add(modelTensorOutput.toBytes());
+            tensorOutputs.add(parseModelTensorOutput(output, resultFilter));
+        }
+        return new ModelTensorOutput(tensorOutputs);
+    }
+
+    protected void doLoadModel(List<Predictor<Input, Output>> predictorList, List<ZooModel<Input, Output>> modelList,
+                               String engine,
+                               Path modelPath,
+                               MLModelConfig modelConfig) throws ModelNotFoundException, MalformedModelException, IOException, TranslateException {
+        tokenizer = HuggingFaceTokenizer.builder().optPadding(true).optTokenizerPath(modelPath.resolve("tokenizer.json")).build();
+        idf = new HashMap<>();
+        if (Files.exists(modelPath.resolve("idf.json"))){
+            Gson gson = new Gson();
```

Comment:
Use gson instance in [StringUtils](https://github.com/opensearch-project/ml-commons/blob/ff0d73460ee03e7f550c9383529a6627f4ac7ed3/common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java#L26) class.

Reply:
Already done it.